packages/phone-number-privacy/combiner/test/unit/domainThresholdState.test.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  DomainQuotaStatusResponseSuccess,
+  DomainRestrictedSignatureResponseSuccess,
+  SequentialDelayDomainState,
+} from '@celo/phone-number-privacy-common'
+import { getVersion } from '@celo/phone-number-privacy-signer/src/config'
+import Logger from 'bunyan'
+import { Request, Response } from 'express'
+import { Session } from '../../src/common/session'
+import config from '../../src/config'
+import { DomainThresholdStateService } from '../../src/domain/services/thresholdState'
+
+describe('domain threshold state', () => {
+  const expectedVersion = getVersion()
+  const domainThresholdStateService = new DomainThresholdStateService(config.domains)
+
+  // TODO: add tests with failed signer responses, depending on
+  // result of https://github.com/celo-org/celo-monorepo/issues/9826
+
+  const getSession = (domainStates: SequentialDelayDomainState[]) => {
+    const mockRequest = {
+      body: {},
+    } as Request
+    const mockResponse = {
+      locals: {
+        logger: new Logger({ name: 'logger' }),
+      },
+    } as Response
+    const session = new Session(mockRequest, mockResponse)
+    domainStates.forEach((status) => {
+      const res: DomainRestrictedSignatureResponseSuccess | DomainQuotaStatusResponseSuccess = {
+        success: true,
+        version: expectedVersion,
+        status,
+      }
+      session.responses.push({ url: 'random url', res, status: 200 })
+    })
+    return session
+  }
+
+  const now = Date.now()
+  const timer = now - 1
+  const counter = 2
+
+  const varyingDomainStates = [
+    {
+      statuses: [
+        { timer, counter: 2, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 2,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer, counter: 1, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 2,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer, counter: 0, disabled: false, now },
+        { timer, counter: 1, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 1,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer, counter: 0, disabled: true, now },
+        { timer, counter: 1, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 2,
```

Comment:
This value is what the current code produces (threshold is 2, the 2nd value of the two enabled states is 2), but shouldn't we be returning 1 in this scenario?